src/main/java/org/radarbase/output/source/S3SourceStorage.kt

```diff
@@ -94,5 +106,29 @@ class S3SourceStorage(
 
     companion object {
         private val logger = LoggerFactory.getLogger(S3SourceStorage::class.java)
+
+        fun <T> faultTolerant(attempt: (Int) -> T): T {
+            var exception: Exception? = null
+            repeat(3) { i ->
+                try {
+                    return attempt(i)
+                } catch (ex: Exception) {
+                    if (ex is ErrorResponseException &&
+                        (ex.errorResponse().code() == "NoSuchKey" || ex.errorResponse().code() == "ResourceNotFound")
+                    ) {
+                        throw FileNotFoundException()
+                    }
+                    logger.warn("Temporarily failed to do S3 operation: {}", ex.toString())
+                    if (i < 2) {
+                        logger.warn("Temporarily failed to do S3 operation: {}, retrying after 1 second.", ex.toString())
+                        Thread.sleep(1000L)
```

Comment:
can we make this exponential? So `2 ^ i`?

Reply:
There's only two sleeps in there. I've made it multiplicative instead.